### PR TITLE
Update to `rkyv`-enabled microkelvin

### LINKED ISSUE #37: Port Nstack to the new Microkelvin
**Describe what you want implemented** 
After the introduction of RKYV in Microkelvin, the most important data-structures need to be migrated as well. 

**Describe "Why" this is needed**
The old version of Microkelvin is incompatible with the new one and therefore nstack should be ported to the new version.

**Describe alternatives you've considered**
N/A

**Additional context**
This entails a simple dependency update of the library. This should be low risk

### PR DESCRIPTION
**Pull request by ureeves**
This ports the NStack to the newest `microkelvin` in the same manner as the Hamt.

Resolves #37

### CODE REVIEW

**Comment by ureeves** [comment]
Blocked awaiting a release of `microkelvin`

File: tests/persist.rs [hunk 1]
```diff
@@ -4,48 +4,103 @@
...
 
-        for i in 0..n {
-            stack.push(i)?;
-        }
+fn persist() -> Result<(), io::Error> {
+    let n: u64 = 1024;
+
+    let mut stack = NStack::<_, Cardinality>::new();
+    for i in 0..n {
+        let i = LittleEndian::from(i);
```
Comment (krl) [inline tests/persist.rs:20]: The endianness is actually not neccesary here.

Remove this line, and remove the `*` - deref below and it works with less complexity.

File: tests/persist.rs [hunk 2]
```diff
@@ -4,48 +4,103 @@
...
+    persist()
+}
 
-        // then empty the original
+fn persist_across_threads() -> Result<(), io::Error> {
+    let n: u64 = 1024;
 
+    let mut stack = NStack::<_, Cardinality>::new();
+    for i in 0..n {
+        let i = LittleEndian::from(i);
```
Comment (krl) [inline tests/persist.rs:65]: same here

File: tests/persist.rs [hunk 3]
```diff
@@ -4,48 +4,103 @@
...
+    // empty original
+    for i in 0..n {
+        assert_eq!(stack.pop().unwrap(), n - i - 1);
     }
+
+    Ok(())
+}
+
+#[test]
+fn persist_across_threads_a() -> Result<(), io::Error> {
```
Comment (krl) [inline tests/persist.rs:89]: We have these concurrency tests in microkelvin already, no need to repeat them here

**Review by krl — CHANGES_REQUESTED** [review]
Looks good, but you can remove the dependency on rend

File: tests/persist.rs (hunk 1 above)
Comment (ureeves) [reply tests/persist.rs]: Removed LittleEndian use in all tests

File: tests/persist.rs (hunk 2 above)
Comment (ureeves) [reply tests/persist.rs]: ditto

File: tests/persist.rs (hunk 3 above)
Comment (ureeves) [reply tests/persist.rs]: Removed

File: tests/nstack.rs [hunk 4]
```diff
@@ -106,66 +100,25 @@ fn branch_lengths() {
...
-        C: Compound<A> + Archive,
-        C::Archived: ArchivedCompound<C, A>,
-        C::Leaf: Archive,
-        A: Annotation<C::Leaf>,
-    {
-        Self {
-            cardinality: Cardinality::combine(anno.clone()),
-            max: MaxKey::combine(anno),
-        }
+    // empty restored copy
```
Comment (krl) [inline tests/nstack.rs:120]: minor nit, this is incorrect, should say something like 'check restored copy'

**Review by krl — APPROVED** [review]

File: tests/nstack.rs (hunk 4 above)
Comment (ureeves) [reply tests/nstack.rs]: You're right, it's not emptying anything. Fixed

**Pull request merged**